pkl-executor/src/main/java/org/pkl/executor/spi/v2/ExecutorSpiOptions2.java

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright © 2024 Apple Inc. and the Pkl project authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.pkl.executor.spi.v2;
+
+import java.net.URI;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+
+public class ExecutorSpiOptions2 {
```

Comment:
For just adding more options, let's make this part of the v1 SPI, but just as an additional class that extends `ExecutorSpiOptions`. This simplifies the SPI stuff quite a bit.

I've prepared a diff that demonstrates the changes that I'm talking about here (not fully tested):

https://gist.github.com/bioball/c1399aabc99e44103408085f0d03133a

Reply:
I don’t think this works. Classes in the spi package are loaded by the class loader that loads pkl-executor classes (see PklDistributionClassloader.loadClass). If you have an old pkl-executor that only has v1.ExecutorSpiOptions, and a newer pkl-core that tries to instantiate v1.ExecutorSpiOptions2, you’ll get a NoClassDefFoundError.

SPI versioning is tricky business. I think it’s best to keep everything strictly separated even if it means some code repetition. (I tried making v2.ExecutorSpiOptions2 extend v1.ExecutorSpiOptions, which I *think* should work, but it didn’t help all that much, and I didn’t feel it was worth the risk and entanglement between versions.) Note that going from 2 to 3 versions will be easier than going from 1 to 2 because I more or less added the infrastructure needed to support n versions.

Reply:
> We need to detail this somewhere, but--PackageResolver isn't a public API.

Currently it is, by definition:
* `EvaluatorBuilder` is a public API. It has a public method `setProjectDependencies(org.pkl.core.project.DeclaredDependencies)` -> `org.pkl.core.project` is a public API.
* `org.pkl.core.project.ProjectDependenciesResolver` has a public constructor that takes `org.pkl.core.packages.PackageResolver` -> `org.pkl.core.packages` is a public API.
* `PackageResolver.getDependencyMetadataAndComputeChecksum` returns `org.pkl.core.util.Pair` -> `org.pkl.core.util` is a public API.

Formalizing the public API by adding support for JPMS (https://github.com/apple/pkl/issues/42) will be a good exercise. It will also likely require breaking API changes. It's something I'd be interested to work on.

Reply:
The way I know this from other projects is that either side should be able to update without breaking the other to avoid a lock-in. If there’s a case that doesn’t work, I’m happy to fix it (and add test coverage) after the build has been stabilized.

Reply:
I'd be happy to merge this PR if it didn't include these SPI changes.

Can you omit the SPI changes from this PR for now? And let's address that in a follow-up PR.

In #227, feel free to add `@Disabled` to tests `evaluate a module that loads a package` and `evaluate a project dependency` so we can stabilize the build without needing to bring in SPI changes.

Reply:
You mean to neither add spi.v2 nor change spi.v1 to pass certificates? Then I’ll also need to disable some tests in this PR.

Reply:
Yeah; no SPI changes for this PR. And yeah, LGTM to disable those tests in this PR too.

Reply:
How about accepting breaking spi.v1 changes for now. That requires far fewer changes, and this needs to be fixed anyway.

Reply:
I know this is higher friction for you, but, I'd prefer to not merge something into main that isn't releasable. If it feels like too much churn for you, I can prepare a version of your PR that excludes the SPI changes (and same with #227)